### ISSUE TITLE
Provide in-app download of elevation data for hill shading

### ISSUE THREAD

**Issue by Lineflyer** [cgeo/cgeo#13604]
> Can you ask or connect me?
> I have roughly 40GB quota free to use (maybe a bit less as the ticket system will keep growing) and the webspace is payed by c:geo anyway.

I don't have any contacts so far and have some stuff going on right now, so if you could reach out that'd be appreciated. Sources below.
3" is - at least for me - the prefered size as the resolution is good enough for geocaching imho and each file (1°x1°) is 2,75MB, so doesn't take too much space. Its also what Locus uses.
Viewfinder 3" is the most complete source, Sonny has better quality but covers only Europe.

| URL                                                                          | Coverage                 | Resolution |
|------------------------------------------------------------------------------|--------------------------|------------|
| https://sonny.4lima.de                                                       | Europe                   | 1", 3"     |
| http://viewfinderpanoramas.org/dem3.html                                     | World                    | 3", 15"    |
| http://viewfinderpanoramas.org/Coverage%20map%20viewfinderpanoramas_org1.htm | USA, Europe, New Zealand | 1"         |

(I don't know how I came to 500MB for Germany before, should be more like 175MB)

_Originally posted by @ztNFny in https://github.com/cgeo/cgeo/issues/13347#issuecomment-1286049415_

**Comment by Lineflyer** [comment]
I have meanwhile contacted the owner of viewfinderpanoramas.org and received his answer as follows:

---

Hello Lars,

Thank you for your interest in the DEM data on my site.

You are welcome to make use of whatever you can find. I do not see any problem with its in-app download by your users.

best regards from Scotland
Jonathan

---

So we can either implement a download from his server or (given that these data does not change frequently) I could ask him for a whole set to put it on a webspace under my (=our) control.

**Comment by moving-bits** [comment]
To write a downloader adapter for this, one needs to write a conversion between latitude/longitude and specific tile file name.
Viewfinderpanoramas has a nice map for manual download: http://www.viewfinderpanoramas.org/Coverage%20map%20viewfinderpanoramas_org3.htm

AFAICS the data is organized in tile files labelled A#-U# for northern hemisphere, and SA#-SQ# for southern hemisphere, with # ranging from 1 to 60, each tile covering 6° of longitude. Not sure about the exact amount of latitude covered by each tile (probably 3°). It looks like "x1" files start at -180° longitude, reaching Greenwich at "x30", and +180° at "x60".

What makes it a bit more complicated, is that some files are aggregated, see Greenland or Antarctica. - But maybe it's ok to ignore those two for Geocaching?

A two-step model could integrate the 1" high resolution data as well. Interactive map is at http://viewfinderpanoramas.org/Coverage%20map%20viewfinderpanoramas_org1.htm, coverage much smaller than at 3". So if the user agrees c:geo could check 1" data first, and download it if available, and switch to 3" data, if not.

As he has agreed to "I do not see any problem with its in-app download by your users" we could stay with download from the original site, at least for the beginning. This would spare us some syncing of available data.

**Comment by moving-bits** [comment]
One additional thing:

Our downloader supports an integrated update check. For this it needs to be able to retrieve a listing including file names and date/time for each file (and preferably size also) - a typical directory listing.

Viewfinderpanoramas seems to have disabled folder listings. If I infer from a download url of `http://viewfinderpanoramas.org/dem3/M18.zip` to folder name `http://viewfinderpanoramas.org/dem3/` and try to download that, I get a "forbidden" error message. 

@Lineflyer 
Can you ask the owner if he would be willing to allow folder listings for `/dem1/ ` and `/dem3/` folders? (Apache supports this on a per-folder base.)

**Comment by ztNFny** [comment]
In case we want to provide the files ourselfes - the files by 4lima.de in my experience are of higher quality for those areas available. But they are hosted at file-hosters, so that'd really require cgeo to host the files themselves.

**Comment by Lineflyer** [comment]
Here is a small update regarding hosting from viewfinderpanoramas.org:

At present there is no quota on my server, although this could be reviewed in the event of a large increase in traffic. I have no objection to you setting up a mirror site. There is about 30GB of data in the 3" files and about the same in the 1" files. The 3" files have not been updated since they were completed in 2012. The coverage of the 1" files is expanding and based on more recent digital ground data.

---

The webspace package I also use for hosting user guide and ticket system has around 40GB left to be used. So it would be doable to host the 3" files (with no need of regular mirroring as those are "stable").
Hosting the 1" files is an alternative but would require regular mirroring as they expand (and can probably hit the 40GB limit if the coverage is widened).
Shall we go with the 3" files of viewfinderpanoramas self hosted or any other preference? I have to admit, that I still did not test the feature myself, so I cannot judge different resolutions or sources.

@ztNFny 
Any idea about the license/usage terms on 4lima.de?

**Comment by ztNFny** [comment]
> Edit: Those are not really suitable as the coverage is only Europe....

yes they are limited, but very up-to-date and high quality for that area. So for Europe - which is together with USA - 90% of the geocaching market - using those makes a lot of sense.

**Comment by SonnyLidarDTM** [comment]
Hi, you can download LiDAR quality elevation files for Europs from https://sonny.4lima.de
Its sizes are 4.8 GB (1"-version) and 0.8 GB (3"-version). You can host them also on c:geo's server
It's license is CC BY 4.0 as mentioned on the Website.
Just for hill shading the 3" version is probably enough. If c:geo wants to display accurate elevation values too, you should better use the 1" version.
